test/mpi/test_barrier.cc

```diff
@@ -73,9 +73,11 @@ void TestBarrier::run()
             if (0 == rank && !completed) {
                 MPI_Test(&rreq, &completed, MPI_STATUS_IGNORE);
             }
+            mpi_progress();
         } while(UCC_OK != status);
-        if  (0 == rank && !completed) {
-            MPI_Wait(&rreq, MPI_STATUS_IGNORE);
+        while (0 == rank && !completed) {
+            MPI_Test(&rreq, &completed, MPI_STATUS_IGNORE);
```

Comment:
need to progress mpi for rank i as well?

Reply:
i think we have enough progress. all other ranks have no place to block and they call mpi_progress from line 76 otherwise

Reply:
ok, I just thought that for the last barrier we need to progress send rank because need to progress ucp_worker

Reply:
the rank that returns from Ssend goes either to the while loop of collective_test (with mpi_progress at line 76) or it would go to next iteration, or to MPI_Barrier which also progresses MPI.